### ISSUE TITLE
setting QBRTCLocalVideoTrack's videoCapture causes screen blink

### ISSUE THREAD

**Issue by RubenBenBen** [QuickBlox/quickblox-ios-sdk#1025]
Hi, I am working on an app where I start a local cameraCapture session **before** calling another user, so I have my own captureSession handler (delegate). It looks like this:
```
-(void)startLocalCapture {
    if (self.cameraCapture) {
        return;
    }
    
    Settings *settings = [Settings instance];
    
    self.cameraCapture = [[BKRTCCameraCapture alloc] initWithVideoFormat:settings.videoFormat
                                                                position:settings.preferredCameraPostion];
    
    [self.cameraCapture startSession:^{
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
            [self.sessionHandler prepareSession:self.cameraCapture.captureSession];
        });
    }];
}
```
and in my session handler:
```
- (void)prepareSession:(AVCaptureSession *)session {
    self.session = session;
    
    for (AVCaptureVideoDataOutput *output in self.session.outputs) {
        self.output = output;
        self.output.connections[0].videoOrientation = AVCaptureVideoOrientationPortrait;
        self.sampleQueue = dispatch_queue_create("sampleQueue", nil);
        [output setSampleBufferDelegate:self queue:self.sampleQueue];
        break;
    }    
}
```
But when I start a call session, I need to set:
```
self.callSession.localMediaStream.videoTrack.videoCapture = self.cameraCapture;
```
which automatically overrides the output's `sampleBufferDelegate` to itself (`QBRTCCameraCapture`). That's ok for me, except that I need to stop the camera session and set its ```sampleBufferDelegate``` to ```nil``` before the line code above to avoid app crash, and then start running the camera session again, and this causes a screen blink:
```
    [self.cameraCapture.session stopRunning];
    [self.output setSampleBufferDelegate:nil queue:self.sampleQueue];
    self.sampleQueue = nil;
    self.callSession.localMediaStream.videoTrack.videoCapture = self.cameraCapture;
    [self..cameraCapture.captureSession startRunning];
```

So basically I will have a screen blink every time I start/end a new call, because I need to override the `sampleBufferDelegate` to keep it working. I have a solution to make handling in a subclass of `QBRTCCameraCapture`, but I need an access to the `(nullable dispatch_queue_t)sampleBufferCallbackQueue` when local captureSession is created.

**Comment by RubenBenBen** [comment]
ok nvm, it actually has an access to the queue.
